lib/util/fields/password_field.dart

```diff
@@ -1,39 +1,114 @@
+  final bool validateStrictly;
 
   const PasswordField({
     super.key,
-    required this.labelText,
-    this.hintText = '',
-    this.errorText,
-    this.helperText,
-    this.controller,
-    this.validator,
-  });
+    required this.controller,
+    this.originalPasswordController,
+    this.validateStrictly = false,
+  }) : assert(validateStrictly == false || originalPasswordController == null);
+
+  @override
+  State<PasswordField> createState() => _PasswordFieldState();
+}
+
+class _PasswordFieldState extends State<PasswordField> {
+  bool _obscureText = true;
 
   @override
   Widget build(BuildContext context) {
     return TextFormField(
+      key: key,
       decoration: InputDecoration(
         border: const OutlineInputBorder(),
         labelText: labelText,
         hintText: hintText,
-        errorText: errorText,
         helperText: helperText,
+        helperMaxLines: 2,
+        suffixIcon: IconButton(
+          icon: Icon(_obscureText ? Icons.visibility : Icons.visibility_off),
+          onPressed: () {
+            setState(() {
+              _obscureText = !_obscureText;
+            });
+          },
+        ),
       ),
-      obscureText: true,
+      obscureText: _obscureText,
       enableSuggestions: false,
       autocorrect: false,
       keyboardType: TextInputType.visiblePassword,
-      controller: controller,
+      controller: widget.controller,
+      onChanged: (_) => setState(() {}),
       validator: validator,
     );
   }
+
+  bool get isConfirmation => widget.originalPasswordController != null;
+
+  String get labelText => isConfirmation ? S.of(context).formPasswordConfirm : S.of(context).formPassword;
+  String get hintText => isConfirmation
+      ? S.of(context).formPasswordConfirmHint
+      : widget.validateStrictly
+          ? S.of(context).formPasswordChooseHint
+          : S.of(context).formPasswordHint;
+
+  Key get key =>
+      widget.key ?? (isConfirmation ? PasswordField.passwordConfirmationFieldKey : PasswordField.passwordFieldKey);
+
+  String? Function(String?) get validator => isConfirmation
+      ? _validateConfirmation
+      : widget.validateStrictly
+          ? _validateStrictly
+          : _validateOnlyEmpty;
+
+  String? get helperText {
+    if (!widget.validateStrictly) return null;
+
+    final String text = widget.controller.text;
+    if (text.isEmpty) return null;
+
+    return _validateStrictly(text);
+  }
+
+  String? _validateConfirmation(String? value) {
+    if (value == null || value.isEmpty) {
+      return S.of(context).formPasswordConfirmValidateEmpty;
+    } else if (value != widget.originalPasswordController?.text) {
+      return S.of(context).formPasswordConfirmValidateMatch;
+    }
+    return null;
+  }
+
+  String? _validateOnlyEmpty(String? value) {
+    if (value == null || value.isEmpty) {
+      return S.of(context).formPasswordValidateEmpty;
+    }
+    return null;
+  }
+
+  String? _validateStrictly(String? value) {
```

Comment:
A general question: an other User Study gave feedback that they prefer to know with the first error what all the requirement for the password are. Should we include this in this PR or do you prefer to leave it as it is?

Reply:
We do have a version of this now don't we? At least you don't have to submit the form to know that the password isn't secure enough yet. I think the current implementation is very cool. Communicating every requirement right away is a bad idea IMO because 1) It overloads the user with information they don't need yet and 2) There is no way to display that really in an acceptable way